src/core/backbone_router/bbr_local.cpp

```diff
@@ -263,6 +263,7 @@ void Local::HandleBackboneRouterPrimaryUpdate(Leader::State aState, const Backbo
     {
         Reset();
     }
+#if !OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
```

Comment:
Not sure if it's a good idea to have different behaviors on reference device and non-reference device. 

Could you give more details on the background? @edmont

Reply:
BBR-TC-02 is failing with OT test bed (it passes with Kirale test bed).

The step 4 forces a BBR Dataset with the maximum Sequence number (255) sent from the BR_2. In the step 6b the assumption is that the BR_2 accepts this Dataset with its RLOC16 but this OT code makes the Sequence Number to increment to 0.

Reply:
Attached failing and passing reports without and with this change in the reference devices.

[BBR-TC-02.zip](https://github.com/openthread/openthread/files/7808388/BBR-TC-02.zip)

Reply:
@edmont 
Thanks for explanation. Make sense to me.